osi_vehicle.proto

```diff
@@ -0,0 +1,172 @@
+    //
+    // \note There may be some constraints how to align the orientation w.r.t.
+    // to some stationary object's or entity's definition.
+    //
+    optional Orientation3d orientation = 5;
+
+    // The relative orientation rate of the moving object w.r.t. its parent
+    // frame and parent orientation rate in the center point of the bounding box
+    // (origin of the bounding box frame).
+    //
+    // <tt>Rotation_yaw_pitch_roll(#orientation (t)) := Rotation_yaw_pitch_roll(#orientation_rate *dt)*Rotation_yaw_pitch_roll(#orientation (t-dt))</tt>
+    //
+    // \note <tt>#orientation (t)</tt> is \b not equal <tt>#orientation (t-dt)+#orientation_rate *dt</tt>
+    //
+    optional Orientation3d orientation_rate = 6;
+
+    // The relative orientation rate acceleration of the moving object w.r.t. its parent
+    // frame and parent orientation rate in the center point of the bounding box
+    // (origin of the bounding box frame).
+    //
+    // <tt>Rotation_yaw_pitch_roll(#orientation (t)) := Rotation_yaw_pitch_roll(#orientation_rate *dt)*Rotation_yaw_pitch_roll(#orientation (t-dt))</tt>
+    //
+    // \note <tt>#orientation (t)</tt> is \b not equal <tt>#orientation (t-dt)+#orientation_rate *dt</tt>
+    //
+    optional Orientation3d orientation_rate_acceleration = 7;
+}
+
+//
+// \brief Interface to the vehicle-model.
+// The focus here is on the powertrain.
+//
+message VehiclePowertrain
+{
+    // The positions of the pedals.
+    //
+    optional Pedalry pedalry = 1;
+    
+    // Rounds per minute of the crankshaft.
+    //
+    optional double engine_rpm = 2;
+
+    // Torque in Nm.
+    //
+    optional double engine_torque = 3;
+
+    // Consumption in liters per 100 km.
+    //
+    optional double engine_consumption = 4;
+
+    // The actual gear of the car.
+    //
+    optional Gear gear = 5;
+}
+
+//
+// \brief Interface to the vehicle-model.
+// The focus here is on the steeringwheel.
+//
+message VehicleSteeringwheel
+{
+    // Angle, angle-speed and torque.
+    // See osi_common_extension.
+    //
+    optional Steeringwheel steeringwheel = 1;
+
+    // Spring-stiffness of the steering in Nm/Deg.
+    //
+    optional double stw_springstiffness = 2;
+
+    // Damping of the steering in Nm*s/Deg.
+    //
+    optional double stw_damping = 3;
+
+    // Friction of the steering in Nm.
+    //
+    optional double stw_friction = 4;
+}
+
+//
+// \brief Interface to the vehicle-model.
+// The focus here is on the wheels.
+// It is made usage of the Wheels-message in osi_common to shorten the code.
+//
+message VehicleWheels
+{
+    // Contains the friction-coefficient of each wheel.
+    // Dimensionless.
+    //
+    optional Wheels friction_coefficient = 1;
+
+    // Contains the z-coordinate (contact-point) of each wheel.
+    // Dimensionless.
+    //
+    optional Wheels contact_point = 2;
+
+    // Contains the rotational speed of each wheel per second.
+    // In [Rad/s].
+    //
+    optional Wheels speed = 3;
```

Comment:
Call it rotational_speed.

Reply:
That is better, I updated the name, thanks!